dev/io.openliberty.microprofile.openapi.ui.internal_fat/fat/src/io/openliberty/microprofile/openapi/ui/internal/fat/tests/UIOauthTest.java

```diff
@@ -0,0 +1,226 @@
+
+        Testcontainers.exposeHostPorts(server.getHttpDefaultPort(), server.getHttpDefaultSecurePort());
+    }
+
+    @Before
+    public void setupTest(){
+        //for some reason the envVars are lost if defined in BeforeClass on the second test despite the fact they don't actually change
+
+        //configure authentication
+        server.addEnvVar(BASIC_AUTH_USERNAME_PROPERTY, BASIC_AUTH_USERNAME);
+        server.addEnvVar(BASIC_AUTH_PASSWORD_PROPERTY, BASIC_AUTH_PASSWORD);
+
+        //Configure Oauth
+        server.addEnvVar(OAUTH_CLIENT_NAME_PROPERTY, OAUTH_CLIENT_NAME);
+        server.addEnvVar(OAUTH_CLIENT_SECRET_PROPERTY, OAUTH_CLIENT_SECRET);
+        server.addEnvVar(OAUTH_REDIRECT_HOST_PROPERTY, HOST);
+
+        // make sure we have a clean driver between tests that can handle self-signed certs
+        driver = new RemoteWebDriver(chrome.getSeleniumAddress(), new ChromeOptions().setAcceptInsecureCerts(true));
+    }
+
+    @After
+    public void teardownTest() throws Exception {
+        //close the browser before stopping the server to reduce changes of open connections
+        driver.quit();
+        server.stopServer();
+    }
+
+    @Test
+    public void defaultPathOAuthTest() throws Exception {
+        // Set the OAuth Endpoint configuration
+        server.addEnvVar(UI_PATH_PROPERTY,UI_PATH_VALUE);
+
+        server.startServer();
+
+        OAuthTest(UI_PATH_VALUE);
+
+    }
+
+    @Test
+    public void customPathOAuthTest() throws Exception{
+        // Set the mpOpenAPI endpoint configuration
+        ServerConfiguration config = server.getServerConfiguration();
+        config.getMpOpenAPIElement().setUiPath(CUSTOM_UI_PATH_VALUE);
+        config.getMpOpenAPIElement().setDocPath(DOC_PATH_VALUE);
+        server.updateServerConfiguration(config);
+        // Set the OAuth Endpoint configuration
+        server.addEnvVar(UI_PATH_PROPERTY, CUSTOM_UI_PATH_VALUE);
+
+        server.startServer();
+
+        OAuthTest(CUSTOM_UI_PATH_VALUE);
+    }
+
+
+    public void OAuthTest(String uiPath){
+        //Need to use HTTPS path for the test for OAuth to actually work
+        driver.get("https://"+HOST+":" + server.getHttpDefaultSecurePort() + uiPath);
+
+        //Store the id of the original window as we need to know when we go through the oauth login flow
+        //We don't need to wait for the page to load to know which one we are after
+        String originalWindow = driver.getWindowHandle();
+
+        // Check the title loads as initial validation page has loaded
+        WebElement title = waitForElement(driver, By.cssSelector("h2.title"), LONG_WAIT);
+        assertThat("Page title", title.getText(), Matchers.containsString("Generated API"));
+
+        //check that the operation is "unlocked"
+        //we will use the lock object later to check that it has been updated
+        WebElement lock = waitForElement(driver, By.cssSelector("button.authorization__btn.unlocked"));
```

Comment:
The CSS selector seems a bit general here. If this is the lock button on the operation, it might be better to find the operation div first and then search within that.

`waitForElement` should work with either `driver` or with a root element to search under.